ERCS/erc-7752.md

```diff
@@ -0,0 +1,937 @@
+```
+
+- **Requirements**:
+  - MUST enforce regulatory compliance for token transfers.
+  - `setCompliance` MUST only be callable by the contract owner.
+  - MUST emit the `ComplianceAdded` event when updated.
+
+### Recovery Mechanisms
+
+The contract SHOULD provide mechanisms to recover tokens if a wallet is lost:
+
+```solidity
+function recoveryAddress(
+    address _lostWallet,
+    address _newWallet,
+    address _investorOnchainID,
+    uint256[] calldata _ids
+) external returns (bool);
+```
+
+- **Requirements**:
+  - MUST only be callable by an address with the `AgentRole`.
+  - MUST verify the identity of the investor through the Identity Registry.
+  - MUST emit the `RecoverySuccess` event upon successful recovery.
+
+### Locking and Unlocking Tokens
+
+The contract MUST allow token holders to lock and unlock their tokens:
+
+```solidity
+function lockTokens(uint256 tokenId, uint256 amount) external;
+function unlockTokens(uint256 tokenId, uint256 amount) external;
+```
+
+- **Requirements**:
+  - MUST prevent locked tokens from being transferred.
+  - MUST emit `TokensLocked` and `TokensUnlocked` events accordingly.
+
+### Events
+
+The contract MUST emit events as specified in the interface to enable off-chain monitoring and compliance:
+
+- `TokensLocked`
+- `TokensUnlocked`
+- `Approval`
+- `ApprovalForAll`
+- `Transfer`
+- `TransferValue`
+- `TokenInvalidated`
+- `URI`
+- `AgentAddedForToken`
+- `AgentRemovedForToken`
+- `UpdatedTokenInformation`
+- `IdentityRegistryAdded`
+- `ComplianceAdded`
+- `Paused`
+- `Unpaused`
+- `Mint`
+- `Burn`
+- `AddressFrozen`
+- `TokenFrozen`
+- `TokenUnfrozen`
+- `RecoverySuccess`
+
+### Access Control
+
+- The contract MUST implement role-based access control to restrict certain functions to authorized addresses (e.g., `AgentRole`).
+- Ownership and agent roles SHOULD be managed securely to prevent unauthorized access.
+
+### Compliance Requirements
+
+- The contract MUST integrate with compliance modules to enforce KYC/AML regulations.
+- Transfers MUST be checked against compliance rules before execution.
+
+### Error Handling
+
+- The contract SHOULD use standard OpenZeppelin error messages and revert reasons for consistency.
+- MUST revert transactions that violate compliance rules, are paused, or involve frozen addresses/tokens.
+
+### Backwards Compatibility
+
+This standard is not backwards compatible with [ERC-20](./eip-20.md) and [ERC-3643](./eip-3643.md) due to:
+
+- Introduction of unique `tokenId` for each issuance or transfer, differentiating tokens beyond simple fungible balances.
+- Replacement of `mint` and `burn` methods with `issue` and `cancel` in earlier drafts, although `mint` and `burn` are used in this interface.
+- Emphasis on individual holdings of securities rather than fungible token balances.
+
+### Notes
+
+- Vesting and governance mechanisms are considered separate concerns and SHOULD be addressed in companion ERCs.
```

Comment:
You cannot create requirements (eg. with "SHOULD") on other ERCs. Try something like:

```suggestion
- Vesting and governance mechanisms are considered separate concerns will be addressed in companion ERCs.
```